File: 11-container-with-most-water/container-with-most-water.py
class Solution(object):
    def maxArea(self, height):
        """
        :type height: List[int]
        :rtype: int
        """
        maxWater= 0
        l , r = 0, len(height) - 1
        while l < r:
            minBar = min(height[l], height[r])
            maxWater = max(maxWater, minBar * (r - l))
            if height[l] < height[r]:
                l += 1
            else:
                r -= 1
        return maxWater

File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    leftWall = 0
    rightWall = height.length - 1
    maxWaterVol = Number.MIN_VALUE
    while(leftWall < rightWall){
        dist = rightWall - leftWall
        waterVol = (Math.min(height[rightWall], height[leftWall])) * dist
        maxWaterVol = Math.max(maxWaterVol, waterVol)
        if(height[leftWall] <= height[rightWall]){
            leftWall += 1
        }
        else{
            rightWall -= 1
        }
    }
    return maxWaterVol
};